sql/core/src/main/scala/org/apache/spark/sql/DataFrame.scala

```diff
@@ -114,7 +114,7 @@ private[sql] object DataFrame {
 @Experimental
 class DataFrame private[sql](
     @transient val sqlContext: SQLContext,
-    @DeveloperApi @transient val queryExecution: SQLContext#QueryExecution) extends Serializable {
+    @DeveloperApi @transient val queryExecution: QueryExecution) extends Serializable {
```

Comment:
@marmbrus This changes the signature of `queryExecution`. Is there anyway that we can preserve the binary compatibility?